pygmt/figure.py

```diff
@@ -340,6 +356,13 @@ def savefig(
             kwargs["Qt"] = 2
             kwargs["Qg"] = 2
 
+        if worldfile:
+            if ext in ["kml", "tiff"]:
```

Comment:
Are worldfiles possible for other formats like pdf/bmp/eps?

Reply:
Yes for PDF and BMP. 

Aslo yes for EPS, but for unknown reasons, the current codes doesn't produce a `.esw` file correctly. Need some time to find out why.

Reply:
Don't think I've ever heard of people create world files for PDF (pfw files?!), but ok then if it works. Have added another suggestion below to test that bmp/bpw files are created.

Reply:
> Are worldfiles possible for other formats like pdf/bmp/eps?

For EPS, see an upstream issue report https://github.com/GenericMappingTools/gmt/issues/7976

Reply:
See https://github.com/GenericMappingTools/gmt/issues/7976, a world file is useless for vector images like EPS and PDF.